Draft: M17 protocol callsign class

turnrye opened the pull request:
This PR refactors to add a class to handle M17 callsigns. Ref https://tasks.openrtx.org/project/openrtx/us/733

TODO: 
- dig in to openrtx/src/rtx/OpMode_M17.cpp to see where types there should just change altogether to the new class rather than instantiating so many objects repeatedly
- review class from a memory management perspective, also what can be made const

silseva commented:
Do you mind changing the code to use `char *` instead of `std::string`? Having `std::string` as a constructor parameter or return value is perfect, but I'd not use it for the internals. This makes the code a bit more simple and, probably, even a bit smaller.

silseva commented on openrtx/src/protocols/M17/Callsign.cpp:
```diff
@@ -0,0 +1,217 @@
+    return true;
+}
+
+const char *Callsign::decode_callsign(const call_t &encodedCall,
```
Also here, we can make it return a `void`

silseva commented on openrtx/include/protocols/M17/Callsign.hpp:
```diff
@@ -0,0 +1,88 @@
+public:
+    static constexpr size_t CALLSIGN_MAX_CHARS = 9;
+
+    Callsign(std::string callsign);
```
Please add headers to these functions

turnrye commented:
Thank you for providing such specific feedback. I've addressed each point, and I added a few unit tests just to make sure that my changes to the existing callsign encoding utility didn't break it. Please give this another lookover and let me know what additional feedback you have. Thanks again!

turnrye commented:
Hey @silseva ! I think I've addressed all of your feedback, please let me know if there are any other things that I need to consider. I also added some naive unit tests since this saw me changing the encoder/decoder behaviors a bit (before this, all I did was move code around). Thank you!
